src/components/Rating/__test__/ratingItems.spec.js

```diff
@@ -10,10 +10,30 @@ describe('<RatingItems />', () => {
             name: 'name-1',
             onChange: expect.any(Function),
             filled: true,
+            half: false,
+            readOnly: false,
         });
     });
     it('should render the right amount of Star components', () => {
         const component = mount(<RatingItems />);
         expect(component.children().length).toBe(5);
     });
+    it('should render half star at the right position when value is a float', () => {
+        const component = mount(<RatingItems value={'3.5'} />);
+        const star = component.childAt(3);
+        expect(star.find('StarHalf').exists()).toBe(true);
+    });
+    it('should not render any half star when value is an integer', () => {
+        const component = mount(<RatingItems value={'5'} />);
+        Array(5)
+            .fill(0)
+            .forEach((item, index) => {
+                expect(
+                    component
+                        .childAt(index)
+                        .find('StarHalf')
+                        .exists(),
+                ).toBe(false);
+            });
+    });
 });
```

Comment:
here we need to add test cases for the filled and half values passed like:
- should pass the right filled when the value passed is float
- should set half to true when the value passed is not integer